plugins/local/non_h_ints_mu/jast_1e_utils.irp.f

```diff
@@ -169,57 +175,70 @@ subroutine get_j1e_coef_fit_ao2(dim_fit, coef_fit)
   ! --- --- ---
   ! get A
 
-  allocate(A(ao_num*ao_num,ao_num*ao_num))
-
-  !$OMP PARALLEL                             &
-  !$OMP DEFAULT (NONE)                       &
-  !$OMP PRIVATE (i, j, k, l, ij, kl, ipoint) &
-  !$OMP SHARED (n_points_final_grid, ao_num, &
-  !$OMP         final_weight_at_r_vector, aos_in_r_array_transp, A)
+  !!$OMP PARALLEL                             &
+  !!$OMP DEFAULT (NONE)                       &
+  !!$OMP PRIVATE (i, j, k, l, ij, kl, ipoint) &
+  !!$OMP SHARED (n_points_final_grid, ao_num, &
+  !!$OMP         final_weight_at_r_vector, aos_in_r_array_transp, A)
+  !!$OMP DO COLLAPSE(2)
+  !do k = 1, ao_num
+  !  do l = 1, ao_num
+  !    kl = (k-1)*ao_num + l
+  !    do i = 1, ao_num
+  !      do j = 1, ao_num
+  !        ij = (i-1)*ao_num + j
+  !        A(ij,kl) = 0.d0
+  !        do ipoint = 1, n_points_final_grid
+  !          A(ij,kl) += final_weight_at_r_vector(ipoint) * aos_in_r_array_transp(ipoint,i) * aos_in_r_array_transp(ipoint,j) &
+  !                                                       * aos_in_r_array_transp(ipoint,k) * aos_in_r_array_transp(ipoint,l)
+  !        enddo
+  !      enddo
+  !    enddo
+  !  enddo
+  !enddo
+  !!$OMP END DO
+  !!$OMP END PARALLEL
+
+  allocate(tmp(ao_num,ao_num,n_points_final_grid))
+  allocate(A(ao_num,ao_num,ao_num,ao_num))
+
+  !$OMP PARALLEL               &
+  !$OMP DEFAULT (NONE)         &
+  !$OMP PRIVATE (i, j, ipoint) &
+  !$OMP SHARED (n_points_final_grid, ao_num, final_weight_at_r_vector, aos_in_r_array_transp, tmp)
   !$OMP DO COLLAPSE(2)
-  do k = 1, ao_num
-    do l = 1, ao_num
-      kl = (k-1)*ao_num + l
-
-      do i = 1, ao_num
-        do j = 1, ao_num
-          ij = (i-1)*ao_num + j
-
-          A(ij,kl) = 0.d0
-          do ipoint = 1, n_points_final_grid
-            A(ij,kl) += final_weight_at_r_vector(ipoint) * aos_in_r_array_transp(ipoint,i) * aos_in_r_array_transp(ipoint,j) &
-                                                         * aos_in_r_array_transp(ipoint,k) * aos_in_r_array_transp(ipoint,l)
-          enddo
-        enddo
+  do j = 1, ao_num
+    do i = 1, ao_num
+      do ipoint = 1, n_points_final_grid
+        tmp(i,j,ipoint) = dsqrt(final_weight_at_r_vector(ipoint)) * aos_in_r_array_transp(ipoint,i) * aos_in_r_array_transp(ipoint,j)
       enddo
     enddo
   enddo
   !$OMP END DO
   !$OMP END PARALLEL
 
-!  print *, ' A'
-!  do ij = 1, ao_num*ao_num
-!    write(*, '(100000(f15.7))') (A(ij,kl), kl = 1, ao_num*ao_num)
-!  enddo
+  call dgemm( "N", "T", ao_num*ao_num, ao_num*ao_num, n_points_final_grid, 1.d0 &
+            , tmp(1,1,1), ao_num*ao_num, tmp(1,1,1), ao_num*ao_num              &
+            , 0.d0, A(1,1,1,1), ao_num*ao_num)
```

Comment:
If you transpose `tmp`, don't forget to change the leading dimension of `tmp` from `ao_num*ao_num` to `n_points_final_grid` (line 221, 2 occurences)

Reply:
Yes (and I will change "N" <---> "T")